### DISCUSSION TITLE
Documentation: how to set up sshfs home file systems

### DISCUSSION THREAD

**Discussion by CharlesMAtkinson** [ltsp/ltsp#284] [Q&A]
I understand from https://ltsp.org/man/ltsp-nfs/ "Note that NFS3 is insecure for home, so by default SSHFS is used" that sshfs is the default for home file systems but I did not find documentation on how to set up sshfs home file systems despite searching for sshfs site:https://ltsp.org and reading all hits including IRC logs less than 3 years old.

**Comment by alkisg** [comment]
Hello, SSHFS is the default, which means that if you just follow the https://ltsp.org/docs/installation/ page you already have it configured without doing anything more.

**Comment by CharlesMAtkinson** [comment]
Thanks alkisg.  I have followed https://ltsp.org/docs/installation/ but do not know where the home directories must be created for them to be mounted by LTSP

**Comment by alkisg** [comment]
You're supposed to use adduser or similar tools directly on the server.

**Comment by CharlesMAtkinson** [comment]
Thanks alkisg. I now understand the LTSP client's user(s) home directories are mounted from the LTSP server's /home/* directories.

ssh password authentication is disabled in the default Debian Buster sshd_config and local policy is to allow only root so I had to edit it:

```
AllowUsers root <space separated list of LTSP client users>
...
Match User <comma separated list of LTSP client users>
  PasswordAuthentication yes
```
I intend to add a page on user management to the LTSP WIKI.  Any advice you want to give before I start?

**Comment by alkisg** [comment]
Hi Charles,

> ssh password authentication is disabled in the default Debian Buster sshd_config

AFAIK password authentication is **enabled** in the default Debian Buster sshd_config.

If you were trying to login as root, this is not a good idea, but if you need it (unrelated to LTSP) it's controlled by the PermitRootLogin option, see `man sshd_config`.

About a user management page in the wiki, sure, it sounds great. Again AFAIK it's exactly the same as user management without LTSP though, i.e. I would just point to the existing wiki pages that desktop environments or distributions have about user management.

**Comment by victor-dolganov** [comment]
Hello! I have two ltsp server installations. One on the desktop computer, and the second on the ubuntu 20.04 server without a UI. In the first, when I enter on client, I see the user's home directory and the desktop settings are saved upon exit and reboot. In the second, I do not see the contents of the user's home directory on the server, all desktop settings are reset to zero when the client is rebooted.
How to check what is missing in the second setup?

**Comment by alkisg** [comment]
Hello victor,

login on a client, and run:

```shell
dpkg -l ltsp sshfs
mount | grep home
````

You should see the ltsp and sshfs versions, and either /home mounted via NFS, or /home/username mounted via SSHFS. If that's not the case, something is definitely wrong.

**Comment by victor-dolganov** [comment]
Thank you! Sshfs was not installed in chroot. Need to update instructions: https://github.com/ltsp/community/wiki/chroots

**Comment by alkisg** [comment]
The instructions seem fine to me, as this command installs sshfs as well as ltsp:

```shell
apt install --install-recommends ltsp
```

**Comment by victor-dolganov** [comment]
root@server:/# history 
    1  echo focal > /etc/hostname
    2  hostname
    3  passwd 
    4  apt install wget
    5  wget https://ltsp.org/misc/ltsp-ubuntu-ppa-bionic.list -O /etc/apt/sources.list.d/ltsp-ubuntu-ppa-bionic.list
    6  wget https://ltsp.org/misc/ltsp_ubuntu_ppa.gpg -O /etc/apt/trusted.gpg.d/ltsp_ubuntu_ppa.gpg
    7  apt update
    8  apt install --install-recommends ltsp
    9  exit
   10  apt install --no-install-recommends linux-generic initramfs-tools
   11  dpkg -l | grep wget
   12  dpkg -l | grep ltsp
   13  apt install xfce4
   14  ls /etc/apt/sources.list
   15  cat /etc/apt/sources.list
   16  cd /etc/apt
   17  ls -l
   18  cat sources.list
   19  apt update
   20  apt install xfce4

I had to copy sources.list from workstation to get all installed.
and sshfs was not installed at 8 step.

**Comment by alkisg** [comment]
Ah indeed, the `apt install --install-recommends ltsp` command is fine, but the commands that manage sources.list are wrong. These `wget` commands are the Debian-way to add a PPA; the Ubuntu-way is to run:

```shell
add-apt-repository ppa:ltsp
add-apt-repository universe
```

Using `add-apt-repository` also ensures that the correct Ubuntu PPA will be used, not always "bionic". So yes please go ahead and update that wiki page; I guess the LTSP users that added that section wanted to avoid installing the `software-properties-common` package, that provides `add-apt-repository`.

**Comment by victor-dolganov** [comment]
And
apt install software-properties-common

Done
